### PR TITLE
Sling 6740 HTL Blog sample

### PR DESCRIPTION
**Pull request by auniverseaway**
This is the HTL blog sample I mentioned in [SLING-6740](https://issues.apache.org/jira/browse/SLING-6740). It's basic by design. In some scenarios it's less DRY to allow for easier understanding for a beginner. The README will have all the details to get a beginner started.

If there's any formatting or syntax constraints I missed, please let me know. If there's anything I should change to match the Sling project, let me know. Of course, any general tips are also appreciated.

Thanks!
Chris

screen-shot

### CODE REVIEW

File: samples/htlblog/pom.xml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    <artifactId>org.apache.sling.samples.htlblog</artifactId>
+    <packaging>bundle</packaging>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>Apache Sling HTL Blog Sample</name>
+    <description>Sample Sling Application - A blog written using HTL.</description>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-scr-plugin</artifactId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: Remove this deprecated plugin and update the bundle plugin to parse the osgi ds annotations

File: samples/htlblog/pom.xml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.sling</groupId>
+        <artifactId>sling</artifactId>
+        <version>22</version>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: you could use the latest version of the parent, it's already at 30

File: samples/htlblog/pom.xml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                    <user>admin</user>
+                    <password>admin</password>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>servlet-api</artifactId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: update servlet api to 3.1

File: samples/htlblog/pom.xml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            <version>2.5</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.compendium</artifactId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: old versions of the osgi libraries

File: samples/htlblog/pom.xml [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        <dependency>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+            <version>1</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.models.api</artifactId>
+            <version>1.1.0</version>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: you could use 1.3 here

File: samples/htlblog/pom.xml [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+        </dependency>
+        <dependency>
+            <groupId>javax.jcr</groupId>
+            <artifactId>jcr</artifactId>
+            <version>2.0</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.jackrabbit</groupId>
+            <artifactId>jackrabbit-jcr-commons</artifactId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: why is this dep needed? you arent using it

File: samples/htlblog/pom.xml [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+            <version>2.11.0</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.jackrabbit</groupId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: why is this dep needed? you arent using it

File: samples/htlblog/pom.xml [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.api</artifactId>
+            <version>2.9.0</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.jcr.api</artifactId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: why is this dep needed? you arent using it

File: samples/htlblog/pom.xml [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.directory.studio</groupId>
+            <artifactId>org.apache.commons.collections</artifactId>
+            <version>3.2.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.scr.annotations</artifactId>
```
Comment (royteeuwen) [inline samples/htlblog/pom.xml]: replace these with the osgi annotations

**Review by royteeuwen — COMMENTED** [review]
I gave some comments about the pom.xml, the rest looks good 👍

**Comment by auniverseaway** [comment]
@royteeuwen Addressed all your comments and added in the needed RAT excludes. The Sling Models API / IMPL on Sling 8 is 1.2.2, so I used that instead of 1.3.

**Comment by royteeuwen** [comment]
Cool, looks nice :)!

**Pull request closed without merging**